server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4796,6 +4815,40 @@ private boolean localDirectoryContains(Directory localDirectory, String file, lo
         return false;
     }
 
+    private void beforeSegmentDownload(RemoteRefreshSegmentTracker downloadStatsTracker, long incomingFileSize) {
+        if (!indexSettings.isRemoteStoreEnabled()) {
+            return;
```

Comment:
Why is this check required? The concerned code only runs for remote store enabled indexes.

Reply:
As discussed, the same `copySegmentFiles` method is being used during snapshot restores from remote store also. This is to handle any indices during snapshot restore which has an overriden index setting to disable RemoteStore. Will add a comment here regarding this.